spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -166,10 +166,12 @@ function onChanOpenInit(
   version: string) {
   // only ordered channels allowed
   abortTransactionUnless(order === ORDERED)
-  // only allow channels to "ibcaccount" port on counterparty chain
-  abortTransactionUnless(counterpartyPortIdentifier === "ibcaccount")
+  // only allow channels to "interchain_account" port on counterparty chain
+  abortTransactionUnless(counterpartyPortIdentifier === "interchain_account")
   // version not used at present
   abortTransactionUnless(version === "ics27-1")
+  // Only open the channel if there is no active channel already set (with status OPEN)
+  abortTransactionUnless(activeChannel === nil)
```

Comment:
This check isn't sufficient. Lets say I create an ICA account. The channel for that account closes, and it becomes inactive. Now there's a race condition between me trying to reopen a channel for that port and someone else sending a `ChanOpenInit` message

There's no check here that prevents any user from submitting a valid `ChanOpenInit` for my port ID except this one. As soon as the channel is inactive, my ICA account is vulnerable 

What is the primary concern with channels closing? From my understanding, the channel can only close if a packet gets timed out. Presumably this should never occur unless a light client attack occurs. If a light client attack occurs, I'd expect the funds in question to have been stolen anyways

Reply:
If a random user manages to call OnChanOpenInit all it will do is open a new channel, it won't give that user access to the account. I think the only reason we need this check is so that we don't have random people creating lots of unnecessary channels. 

I think we can potentially remove the unSet active channel step entirely on OnCloseChan

Reply:
Yes this is correct, after more exploration, the part that was missing for me is that `InitInterchainAccount` should only be called once to create a certain port ID. To recover a ICA after channel closure, the user must send a `MsgChanOpenInit`. I was looking at what would happen if a user didn't initiate a channel handshake via `InitInterchainAccount`